tinygrad/codegen/uops.py

```diff
@@ -36,7 +36,7 @@ def wfxn(*args):
 
 python_alu = {
   UnaryOps.LOG2: lambda x: math.log2(x) if x > 0 else -math.inf if x == 0 else math.nan,
-  UnaryOps.EXP2: hook_overflow(math.inf, lambda x: math.exp(x*math.log(2))),
+  UnaryOps.EXP2: hook_overflow(math.inf, lambda x: 2**x),
```

Comment:
The implementation of `UnaryOps.EXP2` in the Python runtime generated from the example `Tensor([0, 8, 16]).cast(dtypes.uint32)` produces inaccurate results. The results are very close to the correct ones but when casted to `dtypes.uint32` they will be off by one, which is problematic for the implementation of bitcasting as casting has to be accurate. Using Python's power operator directly resolves the issue.